google-cloud-pubsub/lib/google/cloud/pubsub/async_publisher/batch.rb

```diff
@@ -261,16 +257,33 @@ def total_message_bytes
 
           protected
 
-          def items_add msg, callback
-            item = Item.new msg, callback
+          def interval_met? create_time
```

Comment:
Isn't this logic kind of duplicated with the one in AsyncPublisher?
```ruby
              time_since_first_publish = Time.now - @published_at
              if time_since_first_publish > @interval
                # interval met, flush the batches...
                publish_batches!
                @cond.wait
              else
                # still waiting for the interval to publish the batch...
                timeout = @interval - time_since_first_publish
                @cond.wait timeout
              end
```
Now you have to objects that are in charge of the publish interval.